lib/engine/step/tokener.rb

```diff
@@ -49,9 +55,9 @@ def place_token(entity, city, token, teleport: false, special_ability: nil)
         when :neutral
           entity.tokens.delete(token)
           token.corporation.tokens << token
-          @log << "#{entity.name} places a neutral token on #{hex.name}#{price_log}"
+          @log << "#{tokener} places a neutral token on #{hex.name}#{price_log}"
```

Comment:
won't tokener be nil?

Reply:
tokener gets set in either branch of the if statement at line 40. I could set it to “entity.name” before that block if that would be more clear.

Reply:
it doesn't every time

 tokener = "#{entity.name} (#{ability.owner.sym})" if ability.owner != entity

Reply:
How about instead:
```
tokener = entity.name
if ability
  tokener = "#{entity.name} (#{ability.owner.sym})" if ability.owner != entity
  entity.remove_ability(ability)
end
```

Reply:
Oh, right of course. Thanks!